backend/src/tools/memory.py

```
@@ -0,0 +1,60 @@
+		async with get_store_db() as store:
+			documents = await store.asearch(
+				("memories", get_user_id(config)), 
+				query=query, 
+				limit=3, 
+				filter={"user_id": user_id}
+			)
+		print(documents)
+		return [doc.dict() for doc in documents]
+	except ToolException as e:
+		logger.exception(f"Error searching for memories: {e}")
+		raise ToolException(f"Error searching for memories: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)